feat: Add gemini-3-pro-image-preview model support for imageSize para…

choigawoon opened the pull request:
…meters

- Add model identifier to FLASH_IMAGE_PREVIEW_MODEL_IDENTIFIERS
- Add imageSize parameter support (1K, 2K, 4K)
- Fix automatic thinkingConfig injection for image generation models
- Add tests for endpoint routing and parameter transformation
- Update documentation with new model

## Title

<!-- e.g. "Implement user authentication feature" -->

## Relevant issues

<!-- e.g. "Fixes #000" -->
Fixes https://github.com/BerriAI/litellm/issues/17013

## Pre-Submission checklist

**Please complete all items before asking a LiteLLM maintainer to review your PR**

- [x] I have Added testing in the [`tests/litellm/`](https://github.com/BerriAI/litellm/tree/main/tests/litellm) directory, **Adding at least 1 test is a hard requirement** - [see details](https://docs.litellm.ai/docs/extras/contributing_code)
- [x] I have added a screenshot of my new test passing locally 
- [x] My PR passes all unit tests on [`make test-unit`](https://docs.litellm.ai/docs/extras/contributing_code)
- [x] My PR's scope is as isolated as
 possible, it only solves 1 specific problem


<img width="1524" height="998" alt="pr-local-test" src="https://github.com/user-attachments/assets/3d6dd2f8-4654-4bca-994c-3aa39ca4c5eb" />

here`s test-unit result. there are 3 failed but it`s not related with my fixes.
<img width="1461" height="936" alt="image" src="https://github.com/user-attachments/assets/365d2e6b-3c65-4436-a78e-5e9f5c5971f7" />



## Type

<!-- Select the type of Pull Request -->
<!-- Keep only the necessary ones -->

🆕 New Feature
## Changes

krrishdholakia commented:
@choigawoon fixed the thinking config issue with this PR - https://github.com/BerriAI/litellm/pull/17027

can you cleanup your PR to just focus on the imageSize issue?

choigawoon committed:
feat: Add gemini-3-pro-image-preview model support for imageSize parameters

- Add model identifier to FLASH_IMAGE_PREVIEW_MODEL_IDENTIFIERS
- Add imageSize parameter support (1K, 2K, 4K) with GeminiImageSize type
- Add tests for imageSize parameter transformation
- Update documentation with new model

choigawoon commented:
@krrishdholakia 
I’ve updated only the imageSize.
